### ISSUE TITLE
nothing provides requested c_linux-64

### ISSUE THREAD

**Issue by hugobuddel** [conda-forge/conda-forge.github.io#2409]
### Your question:

I'm coming back to conda package maintenance, but I got stuck, because "nothing provides requested c_linux-64". E.g. taking one of the highly starred packages hdf5:

```
docker run -ti continuumio/miniconda3
conda config --add channels defaults
conda config --add channels conda-forge
conda install -y conda-build
git clone https://github.com/conda-forge/hdf5-feedstock.git
conda build hdf5-feedstock
```

leads to

```
Encountered problems while solving:
  - nothing provides requested c_linux-64

Could not solve for environment specs
The following package could not be installed
└─ c_linux-64 does not exist (perhaps a typo or a missing channel).
```

I guess this is related to #2102 , since that is the only issue that turns up when searching for "c_linux-64", but I cannot quickly see what I should do.

How can I get `conda build` to work nowadays? (In particularly, `conda build --dirty`.)

**Comment by hmaarrfk** [comment]
continuumio/miniconda3 is provided by Anaconda.
You can try to use our images.

https://github.com/conda-forge/miniforge-images/

but...... your muscle memory is betraying you!

I typically use

```
python build-locally.py --debug
```
which will drop you in a debug shell after the conda stuff is built out.


cc: @rgommers just for the note on how muscle memory still affects a large part of our community!

**Comment by hmaarrfk** [comment]
> How can I get conda build to work nowadays? (In particularly, conda build --dirty.)

I kinda have trouble with this outside of the `python build-locally.py` stuffs.

I typically comment out the `{{ stdlib('c') }}` line while i debug locally....

**Comment by hugobuddel** [comment]
Thank you @hmaarrfk! 

Commenting out that stdlib line works. Well kinda, it seems that my problem was libc-related, so it 'disappeared'. (I cannot run docker on all my machines, so knowing this workaround is still valuable knowledge for me.)

And now I understand. `c_linux-64` refers to the (right) libc library; and that cannot be provided if it doesn't already exist. I found #2102, but I could not find the string c_linux-64 there (because 1600 items are collapsed), so I did not actually make the connection. Perhaps it is good to explicitly spell out somewhere that `c_linux-64` is related to `stdlib('c')`?

I'll try `python build-locally.py --debug`. I was already hacking around in the `.scripts` directory to get `--dirty` working...

**Comment by hugobuddel** [comment]
I'll close this, as the issue of the title is resolved.

FWIW to future readers, `python build-locally.py --debug` calls `conda debug`, which is similar to `conda build --dirty`, except that it doesn't actually call the build script. `conda debug` was introduced in conda build  3.17.0 (on December 1 2018), so apparently my muscle memory is older than that...

Perhaps it would be useful to explain the `--debug` flag on https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#local-debugging-with-build-locallypy and/or https://conda-forge.org/docs/maintainer/updating_pkgs/#testing-changes-locally?

**Comment by hmaarrfk** [comment]
You have to specify the global conda-build  config file somehow….. not sure.

You used docker in your first example, so I assumed you could use it!

**Comment by hugobuddel** [comment]
Yeah, I can use docker on this machine, and the `build-locally.py --debug` is an excellent solution to do things right. I only mentioned that I sometimes cannot use docker to ensure you that your comment-it-out remark was very useful information to me.

In retrospect, the main thing that went wrong was that I was not aware of the stdlib migration, and that it was piggy backed on other migrations, the rest followed from that.

The cfitsio migration broke one of my packages (https://github.com/conda-forge/cpl-feedstock/pull/20 ). This is usually a trivial migration, so I went into it like "this is probably a 5 minute fix, 'conda build --dirty' should be enough for me to fix it".

But all the problems seem to be created by the stdlib migration that I didn't even realize was going on. It would perhaps have been better (for me) to have a separate migration to glibc 2.17. Then I would have started with a different mindset, (hopefully) would have realized that c_linux-64 was actually about libc, and would probably have tried `build-locally.py` properly.

Now I had to tackle three problems at once: the cfitsio migration, the glibc migration, and me changing my build tools. I'd rather do that one-by-one. (Actually 4 problems: the docker image is changed, and the new one doesn't seem to have perl, so my tests break. Ultimately a good thing though.)

Thanks for your help! Now I still need to fix the actual problems, but that will be tomorrow.
